### PR TITLE
Create directive to handle mobile touches

### PR DESCRIPTION
**Pull request by rjs-picturepark**

### CODE REVIEW

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/directives/mobile-touch.directive.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+})
+export class MobileTouchDirective {
+  @Input() pressTime = 300;
+  @Output() public ppPress: EventEmitter<MouseEvent> = new EventEmitter();
+  @Output() public ppClick: EventEmitter<MouseEvent> = new EventEmitter();
+
+  isLongPress = false;
+  pressTimer: NodeJS.Timeout;
+
+  constructor() {}
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/directives/mobile-touch.directive.ts:14]: remove unnecessary constructor

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/directives/mobile-touch.directive.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@Directive({
+  selector: '[ppMobileTouch]',
+})
+export class MobileTouchDirective {
+  @Input() pressTime = 300;
+  @Output() public ppPress: EventEmitter<MouseEvent> = new EventEmitter();
+  @Output() public ppClick: EventEmitter<MouseEvent> = new EventEmitter();
+
+  isLongPress = false;
+  pressTimer: NodeJS.Timeout;
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/directives/mobile-touch.directive.ts:12]: Change typing to number

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/directives/mobile-touch.directive.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  constructor() {}
+
+  @HostListener('touchstart')
+  public onMouseDown(): void {
+    this.pressTimer = setTimeout(() => {
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/directives/mobile-touch.directive.ts:18]: use window.setTimeout in order to return number

File: src/picturepark-sdk-v1-angular/src/tsconfig.app.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "../out-tsc/app",
-    "types": []
+    "types": ["node"]
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/src/tsconfig.app.json:5]: remove

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-browser/content-browser.component.html [hunk 5]
```diff
@@ -48,13 +48,14 @@
...
+          ppMobileTouch
           [browser]="self"
           [itemModel]="itemModel"
           [thumbnailSize]="activeThumbnailSize"
           [isListView]="activeView.type === 'list'"
           (previewItemChange)="previewItem($event)"
-          (press)="itemPressed($event, i)"
-          (click)="selectedItems.length ? itemClicked($event, i) : previewItem(itemModel)"
+          (ppPress)="itemPressed($event, i)"
+          (ppClick)="selectedItems.length ? itemClicked($event, i) : previewItem(itemModel)"
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/features-module/content-browser/content-browser.component.html:58]: Fix basket interaction

**Review by neotek — APPROVED** [review]

**Pull request merged**